### PR TITLE
Internal: Elements Regression - Use config [ED-8228]

### PR DESCRIPTION
**Pull request by Nevoss**
TODO: 
- [x] Fix snapshot with space at the end
- [x] Try generate elements config dynamicly

### CODE REVIEW

File: tests/elements-regression/lib/controls/control-base.js [hunk 1]
```diff
@@ -72,7 +73,7 @@ class ControlBase {
 	 * @return {string}
 	 */
 	generateSnapshotLabel( value ) {
-		if ( '' === value ) {
+		if ( 'string' === typeof value && '' === value.trim() ) {
```
Comment (matipojo) [inline tests/elements-regression/lib/controls/control-base.js:76]: Explain

File: tests/elements-regression/lib/controls/control-base.js [hunk 2]
```diff
@@ -172,11 +173,9 @@ class ControlBase {
 	 * @return {Promise<void>}
 	 */
 	async clickPopoverToggle() {
-		const popover = await this.getPopover();
+		const toggleControlName = Object.keys( this.config?.condition )[ 0 ].replace( '!', '' );
```
Comment (matipojo) [inline tests/elements-regression/lib/controls/control-base.js:176]: Explain

File: tests/elements-regression/lib/controls/control-base.js [hunk 3]
```diff
@@ -187,6 +186,46 @@ class ControlBase {
 	async getPopover() {
 		return this.page.$( `.elementor-controls-popover:has(${ this.getSelector() })` );
 	}
+
+	/**
+	 * @return {boolean}
+	 */
+	hasConditions() {
+		return isObject( this.config.condition ) || isObject( this.config.conditions );
```
Comment (matipojo) [inline tests/elements-regression/lib/controls/control-base.js:194]: Explain

File: tests/elements-regression/lib/controls/index.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
-	Select,
-	Textarea,
-	Color,
-	Slider,
+	choose: Choose,
+	select: Select,
+	textarea: Textarea,
+	color: Color,
+	slider: Slider,
+	text: Text,
```
Comment (matipojo) [inline tests/elements-regression/lib/controls/index.js:14]: a-z

File: tests/elements-regression/lib/widgets/heading.js [hunk 5]
```diff
@@ -1,16 +1,6 @@
...
-	static getType() {
-		return 'heading';
-	}
-
-	getExcludedControls() {
-		return [
-			'text_stroke_stroke_color', // Cannot be tested alone: setting stroke color without setting stroke width.
-		];
-	}
+const { Widget } = require( './widget' );
```
Comment (matipojo) [inline tests/elements-regression/lib/widgets/heading.js:1]: src

File: tests/elements-regression/playwright.config.js [hunk 6]
```diff
@@ -2,15 +2,16 @@
...
-	reporter: 'list',
-	testDir: '../tests/',
+	reporter: process.env.CI ? 'github' : 'list',
+	testDir: './tests/',
 	globalSetup: require.resolve( './global-setup' ),
 	retries: 1,
 	use: {
 		headless: true,
-		storageState: './tests/playwright/config/storageState.json',
+		storageState: './tests/elements-regression/storage-state.json',
```
Comment (matipojo) [inline tests/elements-regression/playwright.config.js:13]: __dirname

File: tests/elements-regression/tests/elements-regression.test.js-snapshots/.gitignore [hunk 7]
```diff
@@ -1,2 +1,2 @@
-*.jpeg
-!*-linux.jpeg
+*.*
+!*-linux.*
```
Comment (matipojo) [inline tests/elements-regression/tests/elements-regression.test.js-snapshots/.gitignore:2]: Revert

File: tests/elements-regression/tests/elements-regression.test.js [hunk 8]
```diff
@@ -1,90 +1,136 @@
...
-		await editorPage.ensureLoaded();
-		await editorPage.ensureNavigatorClosed();
-		await editorPage.ensureNoticeBarClosed();
-	} );
-
-	test.afterEach( async () => {
-		await wpAdminPage.moveElementorPageToTrash( pageId );
-		await wpAdminPage.deletePermenantlyElementorPageFromTrash( pageId );
+	test.afterAll( async () => {
+		// TODO: If the user use "grep" it will always fail.
```
Comment (matipojo) [inline tests/elements-regression/tests/elements-regression.test.js:17]: Add a summary.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/elements-regression/lib/controls/control-base.js (hunk 1 above)
Comment (Nevoss) [reply tests/elements-regression/lib/controls/control-base.js:76]: Done

File: tests/elements-regression/lib/controls/control-base.js (hunk 2 above)
Comment (Nevoss) [reply tests/elements-regression/lib/controls/control-base.js:176]: Done

File: tests/elements-regression/lib/controls/control-base.js (hunk 3 above)
Comment (Nevoss) [reply tests/elements-regression/lib/controls/control-base.js:194]: Done

File: tests/elements-regression/lib/controls/index.js (hunk 4 above)
Comment (Nevoss) [reply tests/elements-regression/lib/controls/index.js:14]: Done

File: tests/elements-regression/lib/widgets/heading.js (hunk 5 above)
Comment (Nevoss) [reply tests/elements-regression/lib/widgets/heading.js:1]: Done

File: tests/elements-regression/playwright.config.js (hunk 6 above)
Comment (Nevoss) [reply tests/elements-regression/playwright.config.js:13]: Done

File: tests/elements-regression/tests/elements-regression.test.js (hunk 8 above)
Comment (Nevoss) [reply tests/elements-regression/tests/elements-regression.test.js:17]: Task: ED-8292

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Pull request merged**